src/main/proto/ga4gh/genotype_phenotype_service.proto

```diff
@@ -0,0 +1,202 @@
+}
+
+message SearchPhenotypesRequest {
+    // The `PhenotypeAssociationSet` to search. Mandatory
+    string phenotype_association_set_id = 1;
+
+    // Phenotype ID
+    string id = 2;
+
+    // The system should support regular expression matches
+    string description = 3;
+
+    // type
+    OntologyTerm type = 4;
+
+    // terms should be OR'd together. e.g. (severe OR abnormal)
+    repeated OntologyTerm qualifiers = 5;
+
+    // Age of onset
+    OntologyTerm age_of_onset = 6;
+
+    // Specifies the maximum number of results to return in a single page.
+    // If unspecified, a system default will be used.
+    int64 page_size = 7;
+
+    // The continuation token, which is used to page through large result sets.
+    // To get the next page of results, set this parameter to the value of
+    // `next_page_token` from the previous response.
+    string page_token = 8;
+}
+
+// This is the response from `POST /phenotypes/search` expressed as JSON.
+message SearchPhenotypesResponse {
+    // The list of matching PhenotypeInstances.
+    repeated PhenotypeInstance phenotypes = 1;
+
+    // The continuation token, which is used to page through large result sets.
+    // Provide this value in a subsequent request to return the next page of
+    // results. This field will be empty if there aren't any additional results.
+    string next_page_token = 2;
+}
+
+
+/******************  /genotypephenotypes/search  *********************/
+/*
+This request maps to the body of `POST /genotypephenotype/search` as JSON.
+
+The goal here is to allow users to query using one or more of
+Genotype, Phenotype, Environment, and Evidence.
+
+A query using one of the above items is to mean, by default,
+that the remainder of the query is as a "wildcard", such
+that all matches to just that query term would come back.
+Combinations of the above are to act like AND rather than OR.
+
+The "genotype" part of the query methods can be one or more
+genomic features.  Associations can be made at many
+levels of granularity (from whole genotypes down to individual
+SNVs), but users may use these methods with partial or
+inexact information.  Therefore, the query methods must be
+able to support query of some or all of the associated features.
+Furthermore, use of the relationships between genomic features
+means that when querying for a gene, any variants to that
+gene are also returned.  For example, a query with
+BRCA2 would mean that in addition to any direct associations
+to the BRCA2, all associations to sequence variants of BRCA2 would also
+be returned.  Similarly, queries with OntologyTerms should perform
+the subclass closure.
+
+Each query can be made against a string, an array of external
+identifers (such as for gene or SNP ids), ontology term ids, or
+full feature/phenotype/evidence objects.
+*/
+message SearchGenotypePhenotypeRequest {
+  // The `PhenotypeAssociationSet` to search. Mandatory
+  string phenotype_association_set_id = 1;
```

Comment:
In other portions of the proto, a developer can read the comments here to interpret how requests are to be handled. Could we say more about how these fields are handled? Ideally, a developer can integrate based on reading the inline comments alone.